dhis2_android_maps/src/main/java/org/dhis2/maps/views/MapSelectorScreen.kt

```diff
@@ -0,0 +1,454 @@
+    mapSelectorViewModel: MapSelectorViewModel,
+    onBackClicked: () -> Unit,
+    onMapDataUpdated: (FeatureCollection) -> Unit,
+    onLocationButtonClicked: () -> Unit,
+    loadMap: (MapView) -> Unit,
+    onDoneClicked: (result: String?) -> Unit,
+    configurePolygonInfoRecycler: (RecyclerView) -> Unit,
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp),
+        horizontalArrangement = spacedBy(16.dp),
+    ) {
+        Column(
+            modifier = Modifier
+                .weight(0.3f)
+                .fillMaxHeight()
+                .fillMaxSize()
+                .padding(top = 56.dp),
+            verticalArrangement = spacedBy(16.dp),
+        ) {
+            SearchBar(mapSelectorViewModel, onBackClicked)
+
+            LocationInfoContent(mapSelectorViewModel, configurePolygonInfoRecycler)
+
+            DoneButton(mapSelectorViewModel, onDoneClicked)
+        }
+
+        Column(Modifier.weight(0.7f)) {
+            Map(
+                modifier = Modifier
+                    .weight(1f),
+                mapSelectorViewModel = mapSelectorViewModel,
+                onMapDataUpdated = onMapDataUpdated,
+                onLocationButtonClicked = onLocationButtonClicked,
+                loadMap = loadMap,
+            )
+        }
+    }
+}
+
+@Composable
+private fun SearchBar(mapSelectorViewModel: MapSelectorViewModel, onBackClicked: () -> Unit) {
+    val locationItems by mapSelectorViewModel.locationItems.collectAsState(emptyList())
+
+    LocationBar(
+        currentResults = locationItems,
+        onBackClicked = onBackClicked,
+        onClearLocation = mapSelectorViewModel::onClearSearchClicked,
+        onSearchLocation = mapSelectorViewModel::onSearchLocation,
+        onLocationSelected = mapSelectorViewModel::onLocationSelected,
+        onModeChanged = {
+            if (it == SearchBarMode.SEARCH) {
+                mapSelectorViewModel.setCaptureMode(MapSelectorViewModel.CaptureMode.SEARCH)
+            }
+        },
+    )
+}
+
+@Composable
+private fun LocationInfoContent(
+    mapSelectorViewModel: MapSelectorViewModel,
+    configurePolygonInfoRecycler: (RecyclerView) -> Unit,
+) {
+    val selectedLocation by mapSelectorViewModel.selectedLocation.collectAsState()
+
+    val captureMode by mapSelectorViewModel.captureMode.collectAsState()
+
+    when {
+        mapSelectorViewModel.shouldDisplayPolygonInfo() -> {
+            AndroidView(
+                factory = { context ->
+                    RecyclerView(context)
+                },
+                update = configurePolygonInfoRecycler,
+            )
+        }
+
+        captureMode.isNone() && selectedLocation is SelectedLocation.ManualResult -> {
+            with(selectedLocation as SelectedLocation.ManualResult) {
+                LocationItem(
+                    locationItemModel = LocationItemModel.SearchResult(
+                        searchedTitle = "Selected location",
```

Comment:
Could you add the texts in this class to string resources so that it can be translated?